frontend/src/features/admin-form/responses/common/utils/mrfSubmissionView.ts

```diff
@@ -69,3 +69,25 @@ export const getStatusFromWorkflowStatus = (
     }
   }
 }
+
+export type getWorkflowStatusFromFormResponseProps = {
+  index: number
+  workflow: FormWorkflowDto
+  submittedSteps: SubmittedStep[]
+}
+
+/** Gets a step's workflow status from workflow steps and MRF submmission steps */
+export const getWorkflowStatusFromFormResponse = ({
+  index,
+  workflow,
+  submittedSteps,
+}: getWorkflowStatusFromFormResponseProps): WorkflowStatus => {
+  const submittedStep = submittedSteps[index]
+  return index < submittedSteps.length
+    ? workflow[index].approval_field // if this step is an approval step
+      ? 'status' in submittedStep // if status is in submitted step
+        ? submittedStep.status
+        : WorkflowStatus.COMPLETED
+      : WorkflowStatus.COMPLETED
+    : WorkflowStatus.PENDING
```

Comment:
Thanks for splitting this out into a separate function -- it helps to make the code more testable and readable.

One minor note on code complexity -- I've found that nested ternaries look simple but can obscure bugs years down the road. What if we tried expressing it this way:
```suggestion
  // Steps that haven't been submitted must be PENDING
  if (index < submittedSteps.length) {
    return WorkflowStatus.PENDING
  }
  
  // Return the ApprovalStatus of approval steps (either APPROVED or REJECTED)
  if (workflow[index].approval_field && 'status' in submittedStep) {
    return submittedStep.status
  }
  
  // Otherwise, the step is COMPLETED
  return WorkflowStatus.COMPLETED
```

Reply:
That's way way cleaner. 🙏